release/__init__.py

```diff
@@ -565,9 +573,13 @@ def do_build_packages(cache_repository_url):
 
     _build_builders(package_store)
 
-    result = pkgpanda.build.build_tree(package_store, True, None)
+    if is_windows:
+        variants = ['windows']
+    else:
+        variants = [None, 'installer']
```

Comment:
I believe this suggestion came from you. I will try and understand this enough to make an informed fix although I really don't understand this very well.

Reply:
I think I may need some more specific information here so I don't do this wrong.

Reply:
Can you be more specific as to what you want here please.

Reply:
This bit of code is solving a legitimate problem, which is that it no longer makes sense to build all available tree variants, since we now have tree variants that only build on one supported platform or another. So at some point we definitely need to say "if we're on Windows let's build this list of variants, otherwise let's build this different list of variants." However, this decision is currently being made a few function calls deep into `release`. This makes `release` look more platform-dependent than it really is, and it prevents us from layering new tree variants on top of the repo and then building them without modifying build/release tooling (which is currently supported and used). 

I think we can move this decision (and its related platform check) up toward the beginning of the call graph, or even outside the `release` module. Since we have different build entrypoints for Windows and Linux (the separate build scripts that each call the `release` CLI), those seem like the ideal place to state which tree variants are appropriate to build. So I'd propose modifying the `release create` CLI to allow something like this:

```
release create <channel> <tag>  # builds all variants, like normal
release create <channel> <tag> --tree-variant <variant1>  # builds only variant1
release create <channel> <tag> --tree-variant <variant1> --tree-variant <variant2>  # builds only variant1 and variant2
```

This would be implemented by adding a new `tree-variant` argument to the parser for `release create` that uses the `append` action, which would assemble a list of all the `--tree-variant` values (see https://docs.python.org/3/library/argparse.html#action). Note that we'll have to replace `'default'` with `None` if it's in the variant list before we pass it along, since that's how code refers to the default variant internally.

We'd then need to pass that list as a `tree_variants` argument to `ReleaseManager.create()`, which passes it to `make_stable_artifacts()`, which then passes it to `do_build_packages()`. If no `--tree-variant` args are provided on the command line, then we should pass `tree_variants` as `None`, and that should result in all variants getting built. Otherwise `tree_variants` should be a list of variants (e.g. `[None, 'installer']`), and we should only build those variants.

Finally, we'd need to update the Windows and Linux build scripts to build only the appropriate variants: `release create $env:USERNAME local_build --tree-variant windows` and `release create $TAG $TAG --tree-variant default --tree-variant installer`. Client code states directly which tree variants to build, without having to rely on `release` to do a platform check and choose a hardcoded variant list.